openbb_platform/providers/oecd/openbb_oecd/models/gdpforecast.py

```diff
@@ -0,0 +1,73 @@
+from datetime import date
+from typing import Any, Dict, List, Literal, Optional
+
+from openbb_oecd.utils import constants, helpers
+from openbb_provider.abstract.fetcher import Fetcher
+from openbb_provider.standard_models.gdpforecast import (
+    GDPForecastData,
+    GDPForecastQueryParams,
+)
+from pydantic import Field
+
+gdp_countries = tuple(constants.COUNTRY_TO_CODE_GDP_FORECAST.keys())
+
+
+class OECDGDPForecastQueryParams(GDPForecastQueryParams):
+    """GDP Forecast query."""
+
+    country: Literal[*gdp_countries] = Field(
+        description="Country to get GDP for.", default="united_states"
+    )
+
+
+class OECDGDPForecastData(GDPForecastData):
+    """GDP Forecast data from OECD."""
+
+
+class OECDGDPForecastFetcher(
+    Fetcher[OECDGDPForecastQueryParams, List[OECDGDPForecastData]]
+):
+    """OECD GDP Forecast Fetcher."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> OECDGDPForecastQueryParams:
+        if params["start_date"] is None:
+            params["start_date"] = date(1990, 1, 1)
+        if params["end_date"] is None:
+            params["end_date"] = date(date.today().year + 10, 12, 31)
+
+        return OECDGDPForecastQueryParams(**params)
+
+    @staticmethod
+    def extract_data(
+        query: OECDGDPForecastQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> dict:
+        units = query.units[0].upper()
+        if query.type == "real":
+            url = f"https://stats.oecd.org/sdmx-json/data/DP_LIVE/.REALGDPFORECAST.TOT.AGRWTH.{units}/OECD?contentType=csv&detail=code&separator=comma&csv-lang=en&startPeriod={query.start_date}&endPeriod={query.end_date}"
+        elif query.type == "nominal":
+            url = "https://stats.oecd.org/sdmx-json/data/DP_LIVE/.NOMGDPFORECAST.TOT.AGRWTH.{units}/OECD?contentType=csv&detail=code&separator=comma&csv-lang=en&startPeriod={query.start_date}&endPeriod={query.end_date}"
```

Comment:
This should be an f-string. But this can be automated using a private attribute and `model_validator`.

Reply:
whoops i missed the f :sobbing-skull:

Reply:
Press F